### PR TITLE
Assorted build fixes

### PR DESCRIPTION
**Pull request by dlespiau**
I somewhat doubt we've had make distcheck running recently, but this series is a leap towards it.

The (last?) thing remaining is being able to run all docker tests under a non-root user (already captured in an issue).

There's an awkward thing in having the docker tests part of make check (and so make distcheck): it means we need to run have docker running and configured to use the compiled runtime (problably through make install and having docker configured to take the runtime from the installation prefix)

This series also tweaks the travis build to do out-of-tree builds out of an exploded make dist tarball to add extra tests (for instance that we do distribute all the necessary files in the tarball). That's close to what make distcheck does without actually doing a make distcheck (that can't run in travis).

### CODE REVIEW

**Comment by dlespiau** [comment]
Fixes https://github.com/01org/cc-oci-runtime/issues/323

**Comment by dlespiau** [comment]
Fixes https://github.com/01org/cc-oci-runtime/issues/79

**Comment by dlespiau** [comment]
PR updated:

  - Rebased against latest master (was conflicting)
  - Fixed https://github.com/01org/cc-oci-runtime/issues/277
  - make distcheck now passes on my laptop

**Comment by dlespiau** [comment]
Fixes https://github.com/01org/cc-oci-runtime/issues/277
Fixes https://github.com/01org/cc-oci-runtime/issues/322

File: .travis/travis-setup-test-directory.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+# We run the travis script from an exploded `make dist` tarball to ensure
+# `make dist` has the necessary files to compile and run the tests.
+#
+# Note: we don't use `make distcheck` here as we can't run everything we want
+#       for discheck in travis.
+chronic make dist
```
Comment (jodh-intel) [inline .travis/travis-setup-test-directory.sh]: Might be worth a comment saying why `chronic(1)` is being used here.

File: Makefile.am [hunk 2]
```diff
@@ -127,7 +130,8 @@ DISTCHECK_CONFIGURE_FLAGS = \
 	--enable-cppcheck \
 	--enable-valgrind \
 	--enable-code-coverage \
-	--enable-functional-tests
+	--enable-functional-tests \
+	--with-systemdsystemunitdir=/tmp/cc-distcheck/systemdunitdir/
```
Comment (jodh-intel) [inline Makefile.am]: Can this directory be made unique?

Comment (dlespiau) [inline Makefile.am]: Actually, I didn't really want to make it unique: it's not needed (ie. not needing several distinct copies of that data) and doing several make distcheck will not end up clobbering /tmp.

Comment (jodh-intel) [inline Makefile.am]: Maybe a comment to that effect would be worthwhile to explain this.

Comment (dlespiau) [inline Makefile.am]: That I can do!

**Comment by dlespiau** [comment]
Updated PR:
  - Rebased against master (was conflicting with recent developments)
  - Put the build steps inside their own shell script to work around limitations in travis
  - Added some comments about the use chronic
  - Added quoting to $tarball_dir. Note that It's not actually a problem and adding quotes may actually fail to alert us that the dist tarball has a path with a space in, which isn't a nice property for a dist tarball. I'm not going to bikeshed the bikeshed though and just added the quotes.
  - Added a comment about the systemd unit dir in make distcheck

**Comment by grahamwhaley** [comment]
Nice series @dlespiau  I think maybe we are waiting for just one or two tiny addends from @jodh-intel feedback, but other than that,
lgtm

**Comment by dlespiau** [comment]
Fixed the "builddi" typo in commit message.

All the feedback from James should have been addressed, hopefully.

File: .travis.yml [hunk 3]
```diff
@@ -22,13 +22,13 @@ env:
     - coverity_scan_run_condition='"$CC" = gcc'
 
 before_script:
-- ./.travis-setup.sh
+- ./.travis/travis-setup.sh
```
Comment (jodh-intel) [inline .travis.yml]: The leading `./` is redundant, but innocuous :-)

**Comment by jodh-intel** [comment]
lgtm

**Comment by grahamwhaley** [comment]
I don't think the leading ./ will block the merge, so:
lgtm

**Comment by chavafg** [comment]
qa-failed

**Comment by dlespiau** [comment]
Looking at the CI dashboard, it shows that make distcheck passes (yey!) but make check doesn't (from time to time). It seems like it's a known flaky functional test:

```
not ok 42 start and kill state
```

**Comment by jodh-intel** [comment]
surely it's :snowflake: ? :smile:

**Comment by chavafg** [comment]
qa-passed

**Comment by dlespiau** [comment]
@jodh-intel I wish!, ok merging that one. Thanks all for the reviews.

**Pull request merged**